testsuite/tests/typing-misc/typecore_errors.ml

```diff
@@ -0,0 +1,477 @@
+(* TEST
+   * expect
+*)
+
+
+(** Gives an example for every [raise(Error(_,_,_)] in typing/typecore.ml
+    that is no covered by another test in the testsuite and does not
+    require special flags or ppx.
+*)
+
+(** Illegal interval *)
+
+let x = function 'a'..10 -> ()
+[%%expect {|
+Line _, characters 17-24:
+  let x = function 'a'..10 -> ()
```

Comment:
I would prefer the use of a type-correct integer pattern, eg `1 .. 10`, instead of a weird char-int pattern, so that the test would remain robust to a hypothetical reordering of type inference and type-of-intervals checking.